src/components/working-hours/work-shift-dialog.tsx

```diff
@@ -65,66 +68,177 @@ const WorkShiftDialog = ({ workEvents, trucks, workShift, loading, onClose }: Pr
             : null,
       })) ?? [],
     combine: (results) => ({
-      data: results.map((result) => result.data).filter(DataValidation.validateValueIsNotUndefinedNorNull),
+      data: results
+        .map((result) => {
+          const maxTimestamp =
+            result.data?.locations === undefined
+              ? 0
+              : Math.max(...result.data.locations.map((location) => location.timestamp));
+          const minTimestamp =
+            result.data?.locations === undefined
+              ? 0
+              : Math.min(...result.data.locations.map((location) => location.timestamp));
+          return {
+            ...result.data,
+            locations: result.data?.locations ?? [].filter(DataValidation.validateValueIsNotUndefinedNorNull),
```

Comment:
Should be like this, otherwise only the empty array is filtered.
```suggestion
            locations: (result.data?.locations ?? []).filter(DataValidation.validateValueIsNotUndefinedNorNull),
```

Reply:
Resolved